libs/sdk-ui-dashboard/src/presentation/shareDialog/types.ts

```diff
@@ -0,0 +1,52 @@
+// (C) 2019-2021 GoodData Corporation
+import { ComponentType } from "react";
+import { IAccessControlAware } from "@gooddata/sdk-backend-spi";
+import { IAuditableUsers, ObjRef } from "@gooddata/sdk-model";
+import { IShareProps } from "../../types";
+
+/**
+ * @alpha
+ */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface IApplyPayload extends IShareProps {
+    // this payload will have more items,
+}
+
+/**
+ * @alpha
+ */
+export interface IShareDialogProps {
+    /**
+     * Is share dialog visible?
+     */
+    isVisible?: boolean;
+
+    /**
+     * Object to share
+     */
+    sharedObject: IAccessControlAware & IAuditableUsers;
+
+    /**
+     * Current user reference
+     */
+    currentUserRef: ObjRef;
+
+    /**
+     * Callback to be called when user apply share dialog
+     */
+    onApply: (payload: IApplyPayload) => void;
```

Comment:
IApplyPayload sounds too generic. the onApply callback is fine; i suggest to rename the payload type to something more concrete

Reply:
Done